src/auth.ts

```diff
@@ -54,25 +68,19 @@ export async function setAuthorizationHeader(
 
   if (authOptions.masterKey || authOptions.key) {
     const key = authOptions.masterKey || authOptions.key;
-    getAuthorizationTokenUsingMasterKey(verb, resourceId, resourceType, headers, key);
+    getTokenUsingMasterKey(verb, resourceId, resourceType, headers, key);
   } else if (authOptions.resourceTokens) {
     headers[Constants.HttpHeaders.Authorization] = encodeURIComponent(
       getAuthorizationTokenUsingResourceTokens(authOptions.resourceTokens, path, resourceId)
     );
   } else if (authOptions.tokenProvider) {
     headers[Constants.HttpHeaders.Authorization] = encodeURIComponent(
-      await getAuthorizationTokenUsingTokenProvider(authOptions.tokenProvider, {
-        verb,
-        path,
-        resourceId,
-        resourceType,
-        headers
-      })
+      await authOptions.tokenProvider(verb, path, resourceId, resourceType, headers, getTokenUsingMasterKey)
     );
   }
 }
 
-function getAuthorizationTokenUsingMasterKey(
+function getTokenUsingMasterKey(
```

Comment:
nit: I don't like this rename. It's less descriptive. We've got like 3-4 different "tokens" in the SDK, and while this is in the "Auth" file, I still don't like not specifying which token.